staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -737,11 +759,11 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			// "Connection reset by peer" is usually a transient error.
+			// "Connection reset by peer", "Connection refused" or "apiserver is shutting down" are usually a transient errors.
 			// Thus in case of "GET" operations, we simply retry it.
 			// We are not automatically retrying "write" operations, as
 			// they are not idempotent.
-			if !net.IsConnectionReset(err) || r.verb != "GET" {
+			if !net.IsConnectionReset(err) || !IsConnectionRefused(err) || !isAPIServerShutdown(err) || r.verb != "GET" {
```

Comment:
this logic looks off.  If it is a connection reset, then will not a be a connection refused, so this will always execute, right?

Perhaps separate an intermediate bool will help with readability.

Reply:
```
if verb != get{ 
   return err
}

repeatableError := net.IsConnectionReset || isCoinnectionRefused || isapiservershutdown
if !repeatableError{
    return err
}
```

Reply:
yeah, already spotted this in failed test and reworked :-)